manager/integration/tests/test_settings.py

```diff
@@ -183,26 +191,87 @@ def test_setting_toleration_extra():
     Note: `components` in this context is referring to all deployments,
        daemonsets, IM pods, recurring jobs in Longhorn system
     """
-    pass
-
-
-def wait_for_toleration_update(core_api, apps_api, count, set_tolerations):  # NOQA
+    settings = [
+        {
+            "value": "ex.com/foobar:NoExecute;ex.com/foobar:NoSchedule",
+            "expect": [
+                {
+                    "key": "ex.com/foobar",
+                    "value": None,
+                    "operator": "Exists",
+                    "effect": "NoExecute"
+                },
+                {
+                    "key": "ex.com/foobar",
+                    "value": None,
+                    "operator": "Exists",
+                    "effect": "NoSchedule"
+                },
+            ],
+        },
+        {
+            "value": "node-role.kubernetes.io/controlplane=true:NoSchedule",
+            "expect": [
+                {
+                    "key": "node-role.kubernetes.io/controlplane",
+                    "value": "true",
+                    "operator": "Equal",
+                    "effect": "NoSchedule"
+                },
+            ],
+        },
+        {
+            "value": ":",
+            "expect": [
+                {
+                    "key": None,
+                    "value": None,
+                    "operator": "Exists",
+                    "effect": None,
+                },
+            ]
+        },
+        {
+            "value": "",
+            "expect": [],
+        },
+    ]
+
+    chk_removed_tolerations = []
+    for setting in settings:
+        client = get_longhorn_api_client()  # NOQA
```

Comment:
Why do you put this one inside the `for` loop?

Reply:
Seems the update triggers pods to restart and have a new cluster IP assignment. Hence makes the previous client invalid.

Reply:
Thank you for the explanation!